airflow/io/path.py

```diff
@@ -89,15 +83,18 @@ class ObjectStoragePath(CloudPath):
         "_hash",
     )
 
-    def __new__(cls: type[PT], *args: str | os.PathLike, **kwargs: typing.Any) -> PT:
+    def __new__(
+        cls: type[PT],
+        *args: str | os.PathLike,
+        scheme: str | None = None,
```

Comment:
Again this overrides the original function spec. I'm not sure if we want to diverge.

Reply:
This does not. Since both the parent definition and this one take `**kwargs`, adding a keyword-only argument does not change the caller interface at all. The only difference is when `scheme` is not provided by the caller, an explicit argument receives a default (None in this case) while the key would be absent in `**kwargs`. But since the implementation here uses `pop()` anyway, the end result is exactly equivalent.